pre-commit

```diff
@@ -372,6 +372,27 @@ class RepoTreeTest(TestCase):
                     check_var_srcinfo('replaces', package, PACKAGES, line)
                     check_var_srcinfo('groups', package, GROUPS, line)
 
+    def test_repo_depends_valid(self):
+        pkgs = list(list_packages())
+        for package in pkgs:
+            with self.subTest(package=package):
+                self.assertFalse(
+                    lilac_py_has_field(package, 'repo_depends', self),
+                    msg=('package "\033[1;31m{0}\033[m" has repo_depends '
+                         'in lilac.py').format(package))
+                lilac_yaml_path = os.path.join(pkgpath(package), 'lilac.yaml')
+                with git_open(lilac_yaml_path) as lilac_yaml_file:
+                    lilac_yaml = yaml.load(lilac_yaml_file.read(), Loader=yaml.SafeLoader)
+                    for dep in lilac_yaml.get('repo_depends', []):
+                        if isinstance(dep, dict):
+                            dep_package = list(dep.keys())[0]
```

Comment:
My understanding of reading this line is that, this only checks pkgbase name for splitpkgs, which is fine for solving #1414 and should work in most cases (the number of `repo_depends: - pkgbase: pkgname` is limited). Maybe we should leave a note about this for future improvements.

Reply:
Thanks! I added a not about this line in the new commit.